add Docker Desktop 4.35.0 release notes (#20909)
docker/docker.github.io: content/manuals/desktop/release-notes.md

Take a look at the [Docker Public Roadmap](https://github.com/orgs/docker/projects/51/views/1?filterQuery=) to see what's coming next.

## 4.34.0

{{< release-date date="2024-09-26" >}}

{{< desktop-install-v2 all=true beta_win_arm=true version="4.35.0" build_path="/TBD/" >}}

### New

- WIP

### Upgrades

- [Containerd v1.7.21](https://github.com/containerd/containerd/releases/tag/v1.7.21)
- [Docker Engine v27.3.0](https://docs.docker.com/engine/release-notes/27.3/#2730)
- [Docker Scout CLI v1.14.0](https://github.com/docker/scout-cli/releases/tag/v1.14.0)
- Docker Debug `v0.0.37`
- Linux kernel `v6.10.9`

### Bug fixes and enhancements

#### For all platforms

> - Fixed a bug where a proxy in daemon.json would override the proxy set in Docker Desktop.

Reviewer: ```suggestion
- Fixed a bug where proxy settings in `daemon.json` would override proxies set in Docker Desktop settings.
```

> - Fixed bug where some Docker subnet ranges were not able to be used.

Reviewer: ```suggestion
- Fixed a bug where some Docker subnet ranges were not able to be used.
```

- Added Docker VMM, a new powerful backend for creating and managing Docker Desktop Linux VM.
- Available only on Apple Silicon Macs with macOS 12.5+. Docker VMM unleashes blazing performance like never before.

> - Removed `docker-index` as it is now deprecated.

Reviewer: this is referring to an executable binary right? Maybe we should briefly clarify what it was / how it's replaced

Author: I've updated the description, let me know your thoughts!

Reviewer: LGTM

- Fixed a bug where images couldn't be sorted or filtered by tag. Fixes [docker/for-win#14297](https://github.com/docker/for-win/issues/14297).

> - Fixed a bug where the `docker` CLI did not work as expected when the IT admin `registry.json` file was malformed

Reviewer: ```suggestion
- Fixed a bug where the `docker` CLI did not work as expected when the `registry.json` file was malformed.
```

> - Fixed a bug that prevented pushing images from the dashboard. Fixes [docker/for-win#14258](https://github.com/docker/for-win/issues/14258).

Reviewer: ```suggestion
- Fixed a bug where the **Push to Docker Hub** action in the **Images** view would result in an `invalid tag format` error. Fixes [docker/for-win#14258](https://github.com/docker/for-win/issues/14258).
```

Take a look at the [Docker Public Roadmap](https://github.com/orgs/docker/projects/51/views/1?filterQuery=) to see what's coming next.

## 4.34.0

{{< release-date date="2024-09-26" >}}

{{< desktop-install-v2 all=true beta_win_arm=true version="4.35.0" build_path="/TBD/" >}}

### New

> - Docker Desktop Support for Red Hat Enterprise Linux RHEL

Reviewer: ```suggestion
- Support for [Docker Desktop on Red Hat Enterprise Linux](/manuals/desktop/install/linux/rhel.md) is now generally available.
```

- Volume Backup and Share is now generally available and can be found in the **Volumes** tab.

> - Terminal support within Docker Desktop using system shells is now generally available

Reviewer: ```suggestion
- Terminal support within Docker Desktop using system shells is now generally available.
```

> - Enhanced support for IPv6, allowing users to configure their Docker VM for IPv4-only, dual-stack, or IPv6-only operation. See [IPv6 documentation](https://docs.docker.com/go/desktop-ipv6-only)

Reviewer: ```suggestion
- Enhanced support for IPv6, allowing users to configure their Docker VM for IPv4-only, dual-stack, or IPv6-only operation. See [IPv6 documentation](https://docs.docker.com/go/desktop-ipv6-only).
```

> - Improved output of `docker image inspect` to account for multi-platform images (relevant to containerd image store)

Reviewer: ```suggestion
- Improved output of `docker image inspect` to account for multi-platform images  (requires containerd image store).
```

> - Enables users to import and export a subset of multi-platform images

Reviewer: ```suggestion
- Enables users to import and export a subset of multi-platform images (requires containerd image store).
```

> - Support multi-platform images via enabling users to pick a specific platform in `docker history`

Reviewer: ```suggestion
- Support multi-platform images via enabling users to pick a specific platform in `docker history` (requires containerd image store).
```

> - Beta release of Docker VMM - the faster alternative to Apple's Virtualization Framework on MacOS

Reviewer: ```suggestion
- Beta release of Docker VMM - the faster alternative to Apple's Virtualization Framework on MacOS (requires Apple Silicon and macOS 12.5 or later).
```

- Configure CLI completion scripts on Mac

### Upgrades

- [Containerd v1.7.21](https://github.com/containerd/containerd/releases/tag/v1.7.21)
- [Docker Buildx v0.17.1](https://github.com/docker/buildx/releases/tag/v0.17.1)
- [Docker Compose v2.29.7](https://github.com/docker/compose/releases/tag/v2.29.7)
- [Docker Engine v27.3.1](https://docs.docker.com/engine/release-notes/27.3/#2731)
- [Docker Scout CLI v1.14.0](https://github.com/docker/scout-cli/releases/tag/v1.14.0)
- Docker Debug `v0.0.37`
- Linux kernel `v6.10.9`

### Bug fixes and enhancements

#### For all platforms

- Fixed a bug where a proxy in daemon.json would override the proxy set in Docker Desktop.
- Fixed bug where some Docker subnet ranges were not able to be used.
- Added Docker VMM, a new powerful backend for creating and managing Docker Desktop Linux VM.

> - Available only on Apple Silicon Macs with macOS 12.5+. Docker VMM unleashes blazing performance like never before.

Reviewer: Is already under "new"

- Removed `docker-index` as it is now deprecated.
- Fixed a bug where images couldn't be sorted or filtered by tag. Fixes [docker/for-win#14297](https://github.com/docker/for-win/issues/14297).
- Fixed a bug where the `docker` CLI did not work as expected when the IT admin `registry.json` file was malformed
- Fixed a bug that prevented pushing images from the dashboard. Fixes [docker/for-win#14258](https://github.com/docker/for-win/issues/14258).
- Fixed an issue where Docker Desktop startup failed when ICMPv6 setup was not successful.
- Added drivers that allow USB/IP to work.
- Fixed a bug in Enhanced Container Isolation (ECI) [Docker socket mount permissions for derived images](/content/manuals/security/for-admins/hardened-desktop/enhanced-container-isolation/config.md) where it was incorrectly denying Docker socket mounts for some images when Docker Desktop uses the containerd image store.
- Enable `NFT_NUMGEN`, `NFT_FIB_IPV4` and `NFT_FIB_IPV6` kernel modules.

> 

Reviewer: Build UI changes

```suggestion
- Build UI:
  - Highlight build check warnings in the **Completed builds** list.
  - Improve visualization for the build time charts.
  - Image tags added to **Build results** section under the **Info** tab.

```

Take a look at the [Docker Public Roadmap](https://github.com/orgs/docker/projects/51/views/1?filterQuery=) to see what's coming next.

> ## 4.34.0

Reviewer 2: ```suggestion
## 4.35.0
```

Take a look at the [Docker Public Roadmap](https://github.com/orgs/docker/projects/51/views/1?filterQuery=) to see what's coming next.

## 4.34.0

{{< release-date date="2024-09-26" >}}

{{< desktop-install-v2 all=true beta_win_arm=true version="4.35.0" build_path="/TBD/" >}}

### New

- Support for [Docker Desktop on Red Hat Enterprise Linux](/manuals/desktop/install/linux/rhel.md) is now generally available.
- Volume Backup and Share is now generally available and can be found in the **Volumes** tab.
- Terminal support within Docker Desktop using system shells is now generally available.
- Enhanced support for IPv6, allowing users to configure their Docker VM for IPv4-only, dual-stack, or IPv6-only operation. See [IPv6 documentation](https://docs.docker.com/go/desktop-ipv6-only).

> - Containerd image store:

Author: For the new section, opted for simplifying it like this, WDYT @dvdksn ?

Reviewer: Looks good!

  - Improved output of `docker image inspect` to account for multi-platform images.

>   - Enables users to import and export a subset of multi-platform images.

Reviewer: Is this really available?

- https://github.com/docker/cli/pull/5331

Reviewer 2: Yeah, I don't think these are there yet; these will be in v28.0 of the engine.

Reviewer 2: Effectively only changes in Docker Engine were a new flag on the daemon (`--features`), but that cannot be used by Docker Desktop, and an updated version of BuildKit, but we don't know what's in it; https://docs.docker.com/engine/release-notes/27/#2730

- Added drivers that allow USB/IP to work.
- Fixed a bug in Enhanced Container Isolation (ECI) [Docker socket mount permissions for derived images](/manuals/security/for-admins/hardened-desktop/enhanced-container-isolation/config.md) where it was incorrectly denying Docker socket mounts for some images when Docker Desktop uses the containerd image store.
- Enable `NFT_NUMGEN`, `NFT_FIB_IPV4` and `NFT_FIB_IPV6` kernel modules.
- Build UI:
  - Highlight build check warnings in the **Completed builds** list.
  - Improve visualization for the build time charts.
  - Image tags added to **Build results** section under the **Info** tab.
- Improved efficiency of host-side disk utilization for fresh installations on Mac and Linux.
- Fixed a bug that prevented the Sign in enforcement popup to be triggered when token expires.
- Fixed a bug where containers would not be displayed in the GUI immediately after signing in when using [enforced sign-in](/manuals/security/for-admins/enforce-sign-in/_index.md).
- `settings.json` has been renamed to `settings-store.json`

#### For Mac

- Fixed a bug where auto-start containers could be misconfigured after changing filesharing type in settings.

> - Fixed a bug that would cause `~/.docker/cli-plugins` to not be populated on start-up.

Reviewer 3: @lorena, can you add:
- fixed a bug that prevented php composer or postgres to start as non root user https://github.com/docker/for-mac/issues/7415, https://github.com/docker/for-mac/issues/7438

Author: added!

#### For Windows

- New installations of Docker Desktop for Windows now require a Windows version of 19045 or later.
- Fixed an issue that caused a start failure if IPv6 is disabled either in the kernel config or via the kernel command-line in WSL. Fixes [docker/for-win#14240](https://github.com/docker/for-win/issues/14240)
- Fixed the **Clean / Purge data** button on Windows. Fixes [docker/for-win#12650](https://github.com/docker/for-win/issues/14308).
- Disk usage statistics is now displayed in the Dashboard footer installations.
- Improved recovery for WSL distribution issues.

#### For Linux

- Ubuntu 24.04 is now supported on Docker Desktop.

### Known issues

#### For Windows

- Users with versions 4.14.0 and earlier could encounter issues using the in-app update. To update to the latest version, download and install the latest Docker Desktop from this page.

> 

Author: Done!

Take a look at the [Docker Public Roadmap](https://github.com/orgs/docker/projects/51/views/1?filterQuery=) to see what's coming next.

## 4.35.0

{{< release-date date="2024-10-24" >}}

{{< desktop-install-v2 all=true beta_win_arm=true version="4.35.0" build_path="/TBD/" >}}

### New

- Support for [Docker Desktop on Red Hat Enterprise Linux](/manuals/desktop/install/linux/rhel.md) is now generally available.
- Volume Backup and Share is now generally available and can be found in the **Volumes** tab.
- Terminal support within Docker Desktop using system shells is now generally available.
- containerd image store:
  - Improved output of `docker image inspect` to account for multi-platform images.
  - Support multi-platform images via enabling users to pick a specific platform in `docker history`.

> - Beta release of Docker VMM - the faster alternative to Apple's Virtualization Framework on macOS (requires Apple Silicon and macOS 12.5 or later).

Reviewer 4: ```suggestion
- Beta release of Docker VMM - the faster alternative to Apple Virtualization Framework on macOS (requires Apple Silicon and macOS 12.5 or later).
```
To be consistent with the UI?

> - Beta release of Docker VMM - the faster alternative to Apple's Virtualization Framework on macOS (requires Apple Silicon and macOS 12.5 or later).

Reviewer 4: ```suggestion
- Beta release of Docker VMM - the more performant alternative to Apple's Virtualization Framework on macOS (requires Apple Silicon and macOS 12.5 or later).
```
WDYT @aevesdocker?

### Upgrades

- [containerd v1.7.21](https://github.com/containerd/containerd/releases/tag/v1.7.21)
- [Docker Buildx v0.17.1](https://github.com/docker/buildx/releases/tag/v0.17.1)
- [Docker Compose v2.29.7](https://github.com/docker/compose/releases/tag/v2.29.7)
- [Docker Engine v27.3.1](https://docs.docker.com/engine/release-notes/27.3/#2731)
- [Docker Scout CLI v1.14.0](https://github.com/docker/scout-cli/releases/tag/v1.14.0)
- Docker Debug `v0.0.37`
- Linux kernel `v6.10.9`

### Bug fixes and enhancements

#### For all platforms

- Fixed a bug where proxy settings in `daemon.json` would override proxies set in Docker Desktop settings.
- Fixed a bug where some Docker subnet ranges were not able to be used.

> - Added Docker VMM, a new powerful backend for creating and managing Docker Desktop Linux VM.

Reviewer 4: (+ it's for Apple Silicon Macs only)

- Removed [docker-index](https://github.com/docker/index-cli-plugin) as it is now deprecated, you can use `docker scout cves fs://<path to binary>` instead.
- Fixed a bug where images couldn't be sorted or filtered by tag. Fixes [docker/for-win#14297](https://github.com/docker/for-win/issues/14297).
- Fixed a bug where the `docker` CLI did not work as expected when the `registry.json` file was malformed.
- Fixed a bug where the **Push to Docker Hub** action in the **Images** view would result in an `invalid tag format` error. Fixes [docker/for-win#14258](https://github.com/docker/for-win/issues/14258).
- Fixed an issue where Docker Desktop startup failed when ICMPv6 setup was not successful.

> - Added drivers that allow USB/IP to work.

Reviewer 5: @p1-0tr should this mention Linux, Mac, Hyper-V only? If it's a kernel change then WSL 2 users should check their kernel configuration?

Reviewer 6: WSL2 should already support USB/IP out of the box. But, yes, we probably should call out Linux, Mac and Hyper-V, since it is not up to us at to what WSL2 does in this area.

Reviewer 6: `Added drivers, on macOs, Linux and Hyper-V, that allow USB/IP to work.`

- Fixed a bug in Enhanced Container Isolation (ECI) [Docker socket mount permissions for derived images](/manuals/security/for-admins/hardened-desktop/enhanced-container-isolation/config.md) where it was incorrectly denying Docker socket mounts for some images when Docker Desktop uses the containerd image store.

> - Enable `NFT_NUMGEN`, `NFT_FIB_IPV4` and `NFT_FIB_IPV6` kernel modules.

Reviewer 5: These modules are added everywhere except WSL 2. Should we remove from the "For all platforms" section?

Author: mmm but there should we add that line? 🤔  maybe we could do the following?

```suggestion
- Enable `NFT_NUMGEN`, `NFT_FIB_IPV4` and `NFT_FIB_IPV6` kernel modules, for all platforms except WSL
```

else, or can i move them to #### For mac?

- Build UI:
  - Highlight build check warnings in the **Completed builds** list.
  - Improve visualization for the build time charts.
  - Image tags added to **Build results** section under the **Info** tab.
- Improved efficiency of host-side disk utilization for fresh installations on Mac and Linux.
- Fixed a bug that prevented the Sign in enforcement popup to be triggered when token expires.
- Fixed a bug where containers would not be displayed in the GUI immediately after signing in when using [enforced sign-in](/manuals/security/for-admins/enforce-sign-in/_index.md).
- `settings.json` has been renamed to `settings-store.json`

#### For Mac

> - Fixed a bug where mount point permissions were `root:root`, using new virtualization mode.

Reviewer 5: @fredericdalleau does "new virtualisation mode" refer to "Docker VMM"?

Author: I've removed it, let me know if you prefer to put it back

- Fixed a bug where auto-start containers could be misconfigured after changing filesharing type in settings.

> - Use `libkrun` virtualization on Apple Silicon Macs as an experimental feature. Note this is incompatible with Rosetta.

Reviewer 7: This can also be removed, as it is mentioned in the New section
